### PR TITLE
Cancel when cluster have chart-operator helm v2 deployment

### PR DESCRIPTION
**Pull request by tomahawk28**
## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by tomahawk28** [comment]
Tested in ginger 
```
D 04/02 13:18:55 /apis/application.giantswarm.io/v1alpha1/namespaces/ppfz5/apps/chart-operator chartoperatorv1 finding chart-operator deployment as helm v2 in `chart-operator` tenant cluster | app-operator/service/controller/app/v1/resource/chartoperator/create.go:49 | controller=app-operator | event=update | loop=53 | version=19529823
D 04/02 13:18:55 /apis/application.giantswarm.io/v1alpha1/namespaces/ppfz5/apps/chart-operator chartoperatorv1 found chart-operator deployment as helm v2 in `chart-operator` tenant cluster | app-operator/service/controller/app/v1/resource/chartoperator/create.go:52 | controller=app-operator | event=update | loop=53 | version=19529823
```

File: helm/app-operator-chart/templates/rbac.yaml [hunk 1]
```diff
@@ -72,11 +72,12 @@ rules:
   verbs:
   - "create"
 - apiGroups:
-  - extensions
+  - apps
```
Comment (rossf7) [inline helm/app-operator-chart/templates/rbac.yaml:75]: <3

File: helm/app-operator-chart/templates/rbac.yaml [hunk 2]
```diff
@@ -72,11 +72,12 @@ rules:
...
   resources:
   - deployments
   verbs:
-  - 'create'
+  - "create"
```
Comment (rossf7) [inline helm/app-operator-chart/templates/rbac.yaml:79]: Do we need the create? 

I think that was for tiller and I forgot to remove it. 🤦‍♂

File: service/controller/app/v1/resource/chartoperator/create.go [hunk 3]
```diff
@@ -43,6 +45,21 @@ func (r Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return nil
 	}
 
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding chart-operator deployment as helm v2 in %#q tenant cluster", release))
```
Comment (rossf7) [inline service/controller/app/v1/resource/chartoperator/create.go:49]:
```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding %#q deployment in tenant cluster", release))
```

I think message can be simplified and we can keep this change after the v2 to v3 migration.

File: service/controller/app/v1/resource/chartoperator/create.go [hunk 4]
```diff
@@ -43,6 +45,21 @@ func (r Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+		_, err = r.k8sClient.AppsV1().Deployments(key.Namespace(cr)).Get(key.AppName(cr), metav1.GetOptions{})
+		if err == nil {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found chart-operator deployment as helm v2 in %#q tenant cluster", release))
```
Comment (rossf7) [inline service/controller/app/v1/resource/chartoperator/create.go:52]:
```suggestion
			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %#q deployment in tenant cluster", release))
```

File: service/controller/app/v1/resource/chartoperator/create.go [hunk 5]
```diff
@@ -43,6 +45,21 @@ func (r Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
+		} else if apierrors.IsNotFound(err) {
+			// no-op
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("can't find chart-operator deployment as helm v2 in %#q tenant cluster", release))
```
Comment (rossf7) [inline service/controller/app/v1/resource/chartoperator/create.go:60]:
```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find %#q deployment in tenant cluster", release))
```

File: helm/app-operator-chart/templates/rbac.yaml (hunk 2 above)
Comment (tomahawk28) [reply helm/app-operator-chart/templates/rbac.yaml:79]: Oh, then we can delete it!

**Pull request merged**